### PR TITLE
Integration module company

### LINKED ISSUE #2: Integration module company
We need to add the management of facilities (EHPAD) in our API REST to store and manage specific information for each facility. This will help distinguish between different entities and facilitate the management of patients and healthcare professionals within each facility.

## Tasks to be completed
- [x] Create the `company` data model
    - Attributes
        - `id` : Unique identifier for the facility
        - `name` : Name of the facility
        - `address` (Object)
        - `contact` (Object)
        - `siret_number`: SIRET number of the company
        - `approval_number` Approval number (if applicable)
- [x] Create REST Endpoints for `Company`
   - [x] `POST /companies` : Create a new company
   - [x] `GET /companies` : Retrieve a list of all company
   - [x] `GET /companies/{id}` : Retrieve details of a specific company
   - [x] `PUT /companies/{id}`
   - [x] `DELETE /companies/{id}`

- [x] Validation and data security
     - Implement validation to ensure required data is present and properly formatted
     - Secure the endpoints, to ensure only authorized users can create, update, or delete companies
- [x] Unit and Integration Testing
   - Create tests to verify the correct functionality of each endpoint.
   - Ensure validations work correctly and errors are handled appropriately.

## Acceptance Criteria:
- The `Company` model is properly integrated and documented.
- All REST endpoints function as expected and pass unit and integration tests.
- Data validation and security are in place.

### PR DESCRIPTION
**Pull request by NathaelB**
Implementation of routes for company CRUD, for the moment all routes are protected via Keycloak Roles.

Tests have been carried out (15).

### CODE REVIEW

File: apps/company/services/company_service.ts [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+  async create(payload: CreateCompanySchema) {
+    const address = await this.addressService.create(payload.address)
+    const contact = await this.contactService.create(payload.contact)
+
+    const company = await Company.create({
+      ...payload,
+      addressId: address.id,
+      contactId: contact.id,
+    })
+    await company.load('address')
```
Comment (LeadcodeDev) [inline apps/company/services/company_service.ts]: You can use `Promise.all()` instead

File: tests/functional/companies/delete.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Roles } from '#apps/shared/contracts/roles'
+import Professional from '#apps/shared/models/professional'
+import { test } from '@japa/runner'
+
+test.group('Companies delete', () => {
+  test('should sucessfully delete a company with valid id', async ({ client }) => {
+    const professional = await Professional.firstOrFail()
+
+    const response = await client
+      .delete('/v1/companies/9890150790490362881976491')
```
Comment (LeadcodeDev) [inline tests/functional/companies/delete.spec.ts:10]: When you would test, you should to make factory instead seeder.
Then call your factory like `const model = await Factory.make(…)` to create model. instance.

So you can get `model.id` to replace `9890150790490362881976491`

File: tests/functional/companies/update.spec.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { Roles } from '#apps/shared/contracts/roles'
+import Professional from '#apps/shared/models/professional'
+import { test } from '@japa/runner'
+
+test.group('Companies update', () => {
+  test('should sucessfully update a company with valid data', async ({ assert, client }) => {
+    const professional = await Professional.firstOrFail()
+
+    const response = await client
+      .put('/v1/companies/9890153540490362881974927')
```
Comment (LeadcodeDev) [inline tests/functional/companies/update.spec.ts:10]: Same as below

**Review by LeadcodeDev — CHANGES_REQUESTED** [review]

File: tests/functional/companies/delete.spec.ts (hunk 2 above)
Comment (NathaelB) [reply tests/functional/companies/delete.spec.ts:10]: For the moment I've made a simple seeder, maybe in time we'll do something cleaner with Factory.

**Review by LeadcodeDev — APPROVED** [review]

**Pull request merged**